src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1983,7 +1983,7 @@ internal override IConvertibleConversion ClassifyConvertibleConversion(IOperatio
             {
                 if (source.ConstantValue.HasValue && source.ConstantValue.Value is null && destination.IsReferenceType)
```

Comment:
>source.ConstantValue.HasValue && source.ConstantValue.Value is null [](start = 20, length = 67)

I think we should avoid realizing a value for the purpose of this check.

Reply:
Please review references to ConstantValue in our codebase to locate all other similar checks that could be optimized.

---
In reply to: [444576043](https://github.com/dotnet/roslyn/pull/45127#discussion_r444576043) [](ancestors = 444576043)

Reply:
> Please review references to ConstantValue in our codebase to locate all other similar checks that could be optimized.

Done. I've left the test calls alone, as they're testing the codepath, but I've fixed the ones here I saw in the mainline projects (mainly here and in CFGBuilder)

---
In reply to: [444577989](https://github.com/dotnet/roslyn/pull/45127#discussion_r444577989) [](ancestors = 444577989,444576043)